src/statsd_client.cpp

```diff
@@ -59,7 +56,7 @@ inline bool should_send(float sample_rate)
         return true;
     }
 
-    float p = ((float)random() / RAND_MAX);
+    float p = ((float)GetRand(RAND_MAX) / RAND_MAX);
```

Comment:
This is a quite expensive form of random number generator. As I understand the use of the sample rate, it is meant for places where many many calls are done but we're only interested in lets say 10% of samples just to reduce the CPU and network load, so it might be important in this place to have a fast random number generator.

Maybe switch to thread local `FastRandomContext` (they are not thread safe, so we need either locking or thread local instances)?